contrib/google.golang.org/grpc.v12/option.go

```diff
@@ -22,11 +24,11 @@ type InterceptorOption func(*interceptorConfig)
 
 func defaults(cfg *interceptorConfig) {
 	// cfg.serviceName default set in interceptor
-	// cfg.analyticsRate = globalconfig.AnalyticsRate()
+	// cfg.spanOpts = append(cfg.spanOpts, tracer.AnalyticsRate(globalconfig.AnalyticsRate()))
 	if internal.BoolEnv("DD_TRACE_GRPC_ANALYTICS_ENABLED", false) {
-		cfg.analyticsRate = 1.0
+		cfg.spanOpts = append(cfg.spanOpts, tracer.AnalyticsRate(1.0))
 	} else {
-		cfg.analyticsRate = math.NaN()
+		cfg.spanOpts = append(cfg.spanOpts, tracer.AnalyticsRate(math.NaN()))
```

Comment:
This is redundant. Just don't add the option if you don't want analytics.

Reply:
Removed the `else {...tracer.AnalyticsRate(math.NaN())...}`  logic from `defaults` and from `WithAnalytics` + `WithAnalyticsRate` functions.